gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -541,7 +541,7 @@ func (vm *VMKeeper) Call(ctx sdk.Context, msg MsgCall) (res string, err error) {
 				panic(r)
 			default:
 				err = errors.Wrap(fmt.Errorf("%v", r), "VM call panic: %v\n%s\n",
-					r, m.String())
+					r, m.ExceptionsStacktrace())
```

Comment:
Sorry if I said the opposite before, but I've been thinking about this. I think that perhaps it is better to print both the machine string and the stacktrace. The reason being that this message will be logged to stdout/stderr on the machine, so the information in the machine trace might potentially useful for debugging purposes. I have PRs in progress to limit the amount of text that will be produced, so that will no longer be a concern.

Reply:
No worries :) !
What kind of suffix do you think would be good?

Something like that ?
```
rr = errors.Wrap(fmt.Errorf("%v", r), "VM call panic: %v\nMachine State:%s\nStacktrace: %s\n",
					r, m.String(), m.ExceptionsStacktrace())
```

Reply:
Yeah, seems reasonable 👍

Reply:
done here: https://github.com/gnolang/gno/pull/2145/commits/f52079aaf29d9a1f18689a59ce869c7b27ac27b4

Reply:
Re-opening this, since I made more comments anyway:

> The reason being that this message will be logged to stdout/stderr on the machine, so the information in the machine trace might potentially useful for debugging purposes

@deelawn My opinion is that at the time being we're giving an excess of information to the user if we print _both_ the stacktrace and the machine dump.

One of the reasons I was looking forward to having stacktraces was precisely this. The machine dump is generally huge and verbose; and hard to parse unless you know the GnoVM internals.

Here's what I propose, instead:

- Right now, we remove the machine dump here
- We try and see how Gno debugging works, having a stacktrace instead of the machine dump.
- If we can, we try to fix the information that is lacking on the stacktrace directly into it. (For instance, I see inspecting the arguments, even in a limited fashion, to be somewhat useful as long as we make them "safe" to print; we can add this later on)
- If we cannot, and everyone wishes the machine dump to be back, we add it back.

What do you think?

Reply:
After some internal discussion with @thehowl, he had a good idea for getting the most out of both the new stacktrace and the machine string. Let's only print the machine string if this panic originates from the gnovm itself, not a panic statement from within gno code. This can be done by wrapping the panic value in `doOpPanic2` with a custom type, and then doing a type assertion in the keeper. If the type is the custom type we've defined, don't print the machine string. If not, print the machine string. Does this make sense @omarsy?

Reply:
right I think it is better. I will do that

Reply:
done here: https://github.com/gnolang/gno/pull/2145/commits/6e30e4e949eb7c90f425cccc5032460091074829